apps/mobile/src/queries/balance/stacks-balance.query.ts

```diff
@@ -0,0 +1,76 @@
+import { useCallback } from 'react';
+
+import { useMempoolTxsBalance } from '@/queries/stacks/use-mempool-txs-balance';
+import { QueryObserverResult, useQueries } from '@tanstack/react-query';
+
+import { Money, StxCryptoAssetBalance } from '@leather.io/models';
+import {
+  AddressBalanceResponse,
+  createGetStacksAccountBalanceQueryOptions,
+  createStxCryptoAssetBalance,
+  createStxMoney,
+  useCryptoCurrencyMarketDataMeanAverage,
+  useCurrentNetworkState,
+  useStacksClient,
+} from '@leather.io/query';
+import { baseCurrencyAmountInQuote, createMoney, sumMoney } from '@leather.io/utils';
+
+interface StxBalances {
+  totalStxBalance: Money;
+  accounts: QueryObserverResult<StxCryptoAssetBalance, unknown>[];
+}
+
+function useGetStxBalanceByAddresses(addresses: string[]): StxBalances {
+  const { totalData, queries } = useStxBalancesQueries(addresses);
+
+  if (!addresses || addresses.length === 0)
+    return { totalStxBalance: createMoney(0, 'STX'), accounts: [] };
+
+  return {
+    totalStxBalance: totalData,
+    accounts: queries, // TODO as yet unused but perhaps useful
+  };
+}
+
+function useStxBalancesQueries(addresses: string[]) {
```

Comment:
This function is the workhorse to tie together all of the queries. 

I chose to pass in an array of addresses as we can likely refactor this out to the queries package if we want and use it in the extension by passing an array of a single address. 

Ideally I think we can merge the queries it calls into one bigger `useQueries` call but I was unsure to go ahead with that yet or keep things separate. 

Here we need to:
- load mempool data using `useMempoolTxsBalance` 
- `useMempoolTxsBalance` in turn has queries to load confimed transactions `useStacksConfirmedTransactions`

Reply:
I did test manually and it is working but I think I should take some extra time and add more unit tests to make sure. I added some unit tests [here](https://github.com/leather-io/mono/pull/590/files#r1827515351) but I can improve on them

We have no unit tests in the query package for the old code now so my plan was:
- add unit tests for the old code (single account based)
- obtain mocks for `pending` / more transaction types
- make sure the same tests pass with the new code (multi account code)
- further refactor the new code to merge the multiple `useQueries` into one
- either refactor the old code or just migrate `useStxBalancesQueries` to the query package and replace it in the extension also

To get up to speed with this code I had performed [some refactoring](https://github.com/leather-io/mono/pull/542/files#r1822790109) in the query package - in [this PR](https://github.com/leather-io/mono/pull/542) but left it out of here for now as it was bloating the PR and is not actually used at all on mobile. 

I can get working on this task right away, I opened a new linear issue - https://linear.app/leather-io/issue/LEA-1695/increase-test-coverage-for-stx-balances 

Ideally I should have done all of this here but the PR was already really big